fix linting in zendesk theme repo

mm-jpoole opened the pull request:
https://linear.app/maxmind/issue/SCO-7167

linting is broken in main, eslint needs updating and that requires a config file format change. Get it working with as little effort as possible.

## AC

- eslint is updated
- linting works

dhogan8 commented on package.json:
```diff
@@ -2,16 +2,16 @@
     "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-promise": "^6.6.0",
     "eslint-plugin-security": "^3.0.1",
     "eslint-plugin-sort-keys-fix": "^1.1.2",
```
Are we using this plugin anymore?

dhogan8 reviewed (changes requested):
I just want to double check that we're using the following somehow. I don't see them listed in the config anymore.

```
* eslint-plugin-import
* eslint-plugin-json
* eslint-plugin-node
* eslint-plugin-sort-keys-fix
```

mm-jpoole commented:
> I just want to double check that we're using the following somehow. I don't see them listed in the config anymore.
> 
> ```
> * eslint-plugin-import
> * eslint-plugin-json
> * eslint-plugin-node
> * eslint-plugin-sort-keys-fix
> ```

Thanks! Copilot be slacking :rofl: I added back import and sort-keys, removed json cuz it's built in now, removed node because it is unmaintained

dhogan8 reviewed (approved):
Looks great!!
